lib/util.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * @fileoverview Package-private helpers for the installer.
+ */
+
+'use strict'
+
+var cp = require('child_process')
+var fs = require('fs-extra')
+var helper = require('./slimerjs-core')
+var kew = require('kew')
+var path = require('path')
+
+var DEFAULT_CDN = 'https://github.com/graingert/slimer-downloads/releases/download/'
+var libPath = __dirname
+
+/**
+ * Given a lib/location file of a PhantomJS previously installed with NPM,
+ * try to link the binary to this lib/location.
+ * @return {Promise<boolean>} True on success
+ */
+function maybeLinkLibModule(libPath) {
+  return kew.fcall(function () {
+    var libModule = require(libPath)
+    if (libModule.location &&
+        getTargetPlatform() == libModule.platform &&
+        getTargetArch() == libModule.arch) {
+      var resolvedLocation = path.resolve(path.dirname(libPath), libModule.location)
+      if (fs.statSync(resolvedLocation)) {
+        return checkSlimerjsVersion(resolvedLocation).then(function (matches) {
+          if (matches) {
+            writeLocationFile(resolvedLocation)
+            console.log('SlimerJS linked at', resolvedLocation)
+            return kew.resolve(true)
+          }
+        })
+      }
+    }
+    return false
+  }).fail(function () {
+    return false
+  })
+}
+
+/**
+ * Check to make sure a given binary is the right version.
+ * @return {kew.Promise.<boolean>}
+ */
+function checkSlimerjsVersion(slimerPath) {
```

Comment:
we don't need to check the slimer version. We already know it. It's the version of the source code in this directory.